client/status/unrecognized.go

```diff
@@ -4,6 +4,8 @@ import (
 	"github.com/nspcc-dev/neofs-api-go/v2/status"
 )
 
+var errUnrecognizedStatusV2 unrecognizedStatusV2
```

Comment:
// Hmm. One of my comments seem to be missing somehow.

This one should be exported, it's an "unknown" error in its essence, but it still has some code/message/detail according to the definition of NeoFS protocol error.

BTW, I'd unify some of the error interfaces and maybe even unify their implementation, each of them has `status.Status` inside, each can have some `Code()`, `Message()` and `Details()`.

For example (but it'll reworked in nspcc-dev/neo-go#2248): https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/neorpc#Error

Reply:
`errUnrecognizedStatusV2` made exported.

Unification sounds perfect and the these funcs may be usefull. I agree and raise two hands for it. Added them, but with small changes:
```go
type APIErrorDetails interface {
	Code() status.Code
	Message() string
	IterateDetails(f func(*status.Detail) bool)
}
```
We already have `IterateDetails` in basic `status.Status`, decided reuse it. Is it ok to left this `IterateDetails`?